fluidkeys/secretsend.go

```diff
@@ -64,29 +66,25 @@ https://download.fluidkeys.com#` + recipientEmail + `
 		return 1
 	}
 
-	_, err = encryptSecret("dummy data to test encryption", pgpKey)
+	_, err = encryptSecret("dummy data to test encryption", pgpKey, "")
 	if err != nil {
 		printFailed("Couldn't encrypt to the key:")
 		out.Print("Error: " + err.Error() + "\n")
 		return 1
 	}
 
-	out.Print("\n")
-	out.Print(colour.Info(femaleSpyEmoji + "  Type or paste your message, ending by typing Ctrl-D\n"))
-	out.Print(colour.Info("   It will be end-to-end encrypted so no-one else can read it\n\n"))
-
-	secret, err := scanUntilEOF()
-	if err != nil {
-		log.Panic(err)
-		return 1
+	var secret *string
+	if filename != "" {
+		secret, err = getSecretFromFile(filename)
+	} else {
+		secret, err = getSecretFromStdin()
 	}
-
-	if strings.TrimSpace(secret) == "" {
-		printFailed("Exiting due to empty message.\n")
+	if err != nil {
+		printFailed("Error: " + err.Error())
 		return 1
 	}
 
-	encryptedSecret, err := encryptSecret(secret, pgpKey)
+	encryptedSecret, err := encryptSecret(*secret, pgpKey, filename)
```

Comment:
you're gonna want to convert this into *basename* somewhere, e.g. if it's `/tmp/secret.txt`, only encode `secret.txt`

Reply:
You're right!